kiwi/package_manager/apt.py

```diff
@@ -247,42 +253,33 @@ def _process_install_requests_bootstrap(self) -> CommandCallT:
                 '?exact-name(usr-is-merged)',
                 'base-passwd'
             ] + self.package_requests
+            # Download solved bootstrap packages
             result = Command.run(
                 download_bootstrap, self.command_env
             )
             log.debug(
                 'Apt download: {0} {1}'.format(result.output, result.error)
             )
-            script = dedent('''\n
-                set -e
-                while read -r deb;do
-                    echo "Unpacking $deb"
-                    dpkg-deb --fsys-tarfile $deb | tar -C {0} -x
-                done < {1}
-                while read -r deb;do
-                    pushd "$(dirname "$deb")" >/dev/null || exit 1
-                    if [[ "$(basename "$deb")" == base-passwd* ]];then
-                        echo "Running pre/post package scripts for $(basename "$deb")"
-                        dpkg -e "$deb" "{0}/DEBIAN"
-                        test -e {0}/DEBIAN/preinst && chroot {0} bash -c "/DEBIAN/preinst install"
-                        test -e {0}/DEBIAN/postinst && chroot {0} bash -c "/DEBIAN/postinst"
-                        rm -rf {0}/DEBIAN
-                    fi
-                    popd >/dev/null || exit 1
-                done < {1}
-            ''')
-
-            script.format(self.root_dir, package_names.name)
-
-            with open(package_extract.name, 'w') as install:
-                install.write(
-                    script.format(self.root_dir, package_names.name)
+            # Extract bootstrap packages
+            with open(package_names.name) as packages:
+                solved_debootstrap_packages = [p.rstrip() for p in packages]
+            self.command_env['PATH'] = '$PATH:/usr/bin:/bin:/usr/sbin:/sbin'
+            for package in solved_debootstrap_packages:
+                Command.run(
+                    [
+                        'bash', '-c',
+                        'dpkg-deb --fsys-tarfile {0} | tar -C {1} -x'.format(
```

Comment:
Is there a specific reason you're using `tar` here? You can use `dpkg -e <.deb path>` to extract the scripts from a .deb

Reply:
This extraction is not for the scripts. This extraction "installs" the downloaded bootstrap .deb package into the new root-tree. From the man page of dpkg-deb

```
--fsys-tarfile archive
           Extracts the filesystem tree data from a binary package and
           sends it to standard output in tar format.  Together with
           tar(1) this can be used to extract a particular file from a
           package archive.
```

So this call is not related to the script part of things, which comes later.

Reply:
So the process to bootstrap a debian based system follows the following logic

1. apt-get _with_call_options_to_resolve_packages_and_only_download_them
2. dpkg-deb --fsys-tarfile to extract (install) the set of resolved and downloaded packages
3. run the scripts

Reply:
Ah, I see. You could instead just use `dpkg -x <.deb> <dest>`  to skip the intermediate tar stage